jupyterhub/app.py

```diff
@@ -1314,12 +1315,23 @@ def add_url_prefix(prefix, handlers):
         """
     ).tag(config=True)
 
-    default_url = Unicode(
+    default_url = Union(
+        [Unicode(), Callable()],
         help="""
         The default URL for users when they arrive (e.g. when user directs to "/")
 
         By default, redirects users to their own server.
-        """
+
+        Can be a Unicode string (e.g. '/hub/home') or a callable based on the handler object:
+
+            def default_url_fn(handler):
+                user = handler.current_user
+                if user and user.admin:
+                    return '/hub/admin'
+                return '/hub/home'
+
+            c.JupyterHub.default_url = default_url_fn
+        """,
```

Comment:
This section seem to be giving this warning (treated like an error). Perhaps it is because of the code example indentation?

```
Warning, treated as error:
/home/circleci/project/jupyterhub/app.py:docstring of jupyterhub.app.JupyterHub.default_url:10:Unexpected indentation.
Makefile:64: recipe for target 'html' failed
make: *** [html] Error 2
```